extensions/3rdparty/LyricWiki/LyricFind/tests/LyricFindTrackingServiceTest.php

```diff
@@ -102,10 +102,12 @@ public function testTrackResponseCode($amgId, $apiResponse, $res) {
 		$respMock = is_array($apiResponse) ? json_encode($apiResponse) : $apiResponse;
 
 		$this->mockStaticMethod('Http', 'request', $respMock);
-		$this->mockGlobalVariable('wgTitle', $this->mockClassWithMethods('Title', [
+
+		$titleMock = $this->createConfiguredMock( Title::class, [
 			'getArticleID' => 666,
-			'getText' => 'Paradise_Lost:Forever_Failure'
-		]));
+			'getText' => 'Paradise_Lost:Forever_Failure',
+		] );
+		$this->mockGlobalVariable('wgTitle', $titleMock );
 
 		$service = new LyricFindTrackingService();
 		$status = $service->track($amgId, 0, $this->app->wg->Title);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)